zipkin/include/zipkin/tracer.h

```diff
@@ -40,6 +40,41 @@ class Reporter {
   virtual bool flushWithTimeout(std::chrono::system_clock::duration timeout) {
     return true;
   }
+
+  /**
+   * @return the number of spans that have been dropped since this value was
+   * last cleared.
+   */
+  virtual uint64_t droppedSpanCount() const { return 0; }
+
+  /**
+   * In addition to getting the drop count, resets the drop count.
+   * @return the number of spans that have been dropped since this value was
+   * last reset
+   */
+  virtual uint64_t getAndResetDroppedSpanCount() { return 0; }
+
+  /**
+   * @return the period of time to wait between auto flushes
+   */
+  virtual SteadyClock::duration reportPeriod() const {
+    return SteadyClock::duration::zero();
+  }
+
+  /**
+   * @param report_period the period of time to wait between auto flushes
+   */
+  virtual void setReportPeriod(SteadyClock::duration report_period){};
+
+  /**
+   * @return the number of spans that can be stored in the buffer
+   */
+  virtual size_t bufferSpanCount() const { return 0; }
```

Comment:
Could we rename `bufferSpanCount` to `maxBufferedSpans` or `spanBufferSize`? -- the name doesn't seem quite right.